pgxn/neon/control_plane_connector.c

```diff
@@ -236,42 +238,68 @@ SendDeltasToControlPlane()
 	if (!ForwardDDL)
 		return;
 
+	if (handle == NULL)
+	{
+		static struct curl_slist *headers = NULL;
+
+		headers = curl_slist_append(headers, "Content-Type: application/json");
+		if (headers == NULL)
+		{
+			elog(ERROR, "Failed to set Content-Type header");
+		}
+
+		if (jwt_token)
+		{
+			char		auth_header[8192];
+
+			snprintf(auth_header, sizeof(auth_header), "Authorization: Bearer %s", jwt_token);
+			headers = curl_slist_append(headers, auth_header);
+			if (headers == NULL)
+			{
+				elog(ERROR, "Failed to set Authorization header");
+			}
+		}
+
+		handle = alloc_curl_handle();
+
+		curl_easy_setopt(handle, CURLOPT_CUSTOMREQUEST, "PATCH");
+		curl_easy_setopt(handle, CURLOPT_HTTPHEADER, headers);
+		curl_easy_setopt(handle, CURLOPT_URL, ConsoleURL);
+		curl_easy_setopt(handle, CURLOPT_ERRORBUFFER, CurlErrorBuf);
+		curl_easy_setopt(handle, CURLOPT_TIMEOUT, 3L /* seconds */ );
+		curl_easy_setopt(handle, CURLOPT_WRITEFUNCTION, ErrorWriteCallback);
+	}
+
 	char	   *message = ConstructDeltaMessage();
-	ErrorString str = {};
+	ErrorString str = { 0 };
```

Comment:
This will zero the whole 1kB buffer.  Not a big deal in the big picture, and not actually new with this PR, this just makes it more explicit, but it's probably not intentional. Maybe:
```
ErrorString str;
str.size = 0;
```

Reply:
I did it because the other way is a GNU extension. This really should have been a different commit, so I can split it out to whatever we want. I agree that we could just set size to 0. Let me know if that is what you want.